arwen/elrondapi/manBufOps.go

```diff
@@ -393,3 +399,28 @@ func v1_4_mBufferFinish(context unsafe.Pointer, mBufferHandle int32) int32 {
 	metering.UseGas(gasToUse)
 	return 0
 }
+
+//export v1_4_mBufferNewRandom
+func v1_4_mBufferNewRandom(context unsafe.Pointer, length int32) int32 {
+	managedType := arwen.GetManagedTypesContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+	metering := arwen.GetMeteringContext(context)
+
+	if length < 1 {
+		arwen.WithFault(arwen.ErrLengthOfBufferNotCorrect, context, runtime.ManagedBufferAPIErrorShouldFailExecution())
+		return -1
+	}
+
+	baseGasToUse := metering.GasSchedule().ManagedBufferAPICost.MBufferNewFromBytes
+	lengthDependentGasToUse := math.MulUint64(metering.GasSchedule().BaseOperationCost.DataCopyPerByte, uint64(length))
+	metering.UseGas(baseGasToUse + lengthDependentGasToUse)
+
+	randomizer := managedType.GetRandReader()
```

Comment:
one more change for getRandReader - make it use previous block randomness, current block randomness and tx hash